File: DrinkItUpWebApp/DrinkItUpWebApp/Models/IngredientModel.cs
using DrinkItUpBusinessLogic.DTOs;

namespace DrinkItUpWebApp.Models
{
    public class IngredientModel
    {
        public int IngredientId { get; set; }

        public string Name { get; set; } = null!;

        public decimal Quantity { get; set; }

        public UnitModel Unit { get; set; } = null!;

        public bool IsUsed { get; set; }

        public List<IngredientModel> IngredientsWithUnits { get; set;}
    }
}

File: DrinkItUpWebApp/DrinkItUpSPA/src/app/shared/Models/ingredient.model.ts
import { UnitModel } from "./unit.model";

export class IngredientModel{
    constructor(
        public unit: UnitModel,
        public unitId?: number,
        public ingredientId?: number,
        public name?: string,
        public quantity?: number,
        public isUsed?: boolean
    ){}
}